mail_history/models/mail_message.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class Message(models.Model):
+    _inherit = 'mail.message'
+
+    @api.multi
+    def message_format(self):
+        vals = super().message_format()
+        read_notifs = self.env['mail.notification'].sudo().search(
+            [
+                ('mail_message_id', 'in', self.ids),
+                ('res_partner_id', '!=', False),
+                ('is_read', '=', True),
+            ]
+        )
+        # Obtain partners per message
+        notif_dict = {}
+        for notif in read_notifs:
+            mid = notif.mail_message_id.id
+            notif_dict.setdefault(mid, [])
+            notif_dict[mid].append(
+                notif.res_partner_id.id
+            )
+        # Fill history partner ids field
+        for message in vals:
+            if message['id'] in notif_dict:
+                message.update({
+                    'history_partner_ids': notif_dict[
+                        message['id']
+                    ],
+                })
+        return vals
+
+    @api.model
+    def mark_all_as_read(self, channel_ids=None, domain=None):
+        """Override this method for injecting a context key for avoiding the removal of
+        mail.notification. We need to inject as well a true leaf in the domain for
+        avoiding a part of code in super that removes the records by sql.
+        """
+        if not domain:
+            domain = [['id', '!=', False]]
```

Comment:
Is this necessary?

Reply:
Yes, for avoiding a raw SQL that is executed if not upstream. There should be a docstring/comment talking about this.

Reply:
As I explain in the docstring, We need to add a True leaf in the domain to avoid the deletion via SQL on the super method mark_all_as_read()

Reply:
:+1: